ext/standard/mail.c

```diff
@@ -68,21 +68,18 @@ typedef enum {
 	CONTAINS_NULL
 } php_mail_header_value_error_type;
 
-static php_mail_header_value_error_type php_mail_build_headers_check_field_value(zval *val)
+static php_mail_header_value_error_type php_mail_build_headers_check_field_value(const zend_string *value)
 {
-	size_t len = 0;
-	zend_string *value = Z_STR_P(val);
-
 	/* https://tools.ietf.org/html/rfc2822#section-2.2.1 */
 	/* https://tools.ietf.org/html/rfc2822#section-2.2.3 */
-	while (len < value->len) {
-		if (*(value->val+len) == '\r') {
-			if (*(value->val+len+1) != '\n') {
+	for (size_t len = 0; len < ZSTR_LEN(value); len++) {
```

Comment:
This change is wrong.
Now you always increment len, even if the "continue" is hit. So the following code block:
```c
len += 3;
continue;
```

will now increment by 4 instead of 3.